src/main/java/org/jitsi/videobridge/rest/JSONDeserializer.java

```diff
@@ -164,54 +165,85 @@ public static void deserializeCandidates(
             // lastN
             if (lastN != null)
             {
-                Integer i;
-
-                if (lastN instanceof Integer)
-                    i = (Integer) lastN;
-                else if (lastN instanceof Number)
-                    i = Integer.valueOf(((Number) lastN).intValue());
-                else
-                    i = Integer.valueOf(lastN.toString());
+                Integer i = objectToInteger(lastN);
                 channelIQ.setLastN(i);
             }
             // receivingSimulcastStream
             if (receivingSimulcastStream != null)
             {
-                Integer i;
-
-                if (receivingSimulcastStream instanceof Integer)
-                    i = (Integer) receivingSimulcastStream;
-                else if (receivingSimulcastStream instanceof Number)
-                    i = Integer.valueOf(
-                            ((Number) receivingSimulcastStream).intValue());
-                else
-                    i = Integer.valueOf(receivingSimulcastStream.toString());
+                Integer i = objectToInteger(receivingSimulcastStream);
                 channelIQ.setReceivingSimulcastLayer(i);
             }
             // payloadTypes
             if (payloadTypes != null)
+            {
                 deserializePayloadTypes((JSONArray) payloadTypes, channelIQ);
+            }
             // rtpLevelRelayType
             if (rtpLevelRelayType != null)
+            {
                 channelIQ.setRTPLevelRelayType(rtpLevelRelayType.toString());
+            }
             // sources
             if (sources != null)
+            {
                 deserializeSources((JSONArray) sources, channelIQ);
+            }
             // source groups
             if (sourceGroups != null)
+            {
                 deserializeSourceGroups((JSONArray) sourceGroups, channelIQ);
+            }
             // ssrcs
             if (ssrcs != null)
+            {
                 deserializeSSRCs((JSONArray) ssrcs, channelIQ);
+            }
             // header extensions
             if (headerExtensions != null)
-                deserializeHeaderExtensions((JSONArray) headerExtensions, channelIQ);
+            {
+                deserializeHeaderExtensions(
+                        (JSONArray) headerExtensions,
+                        channelIQ);
+            }
 
             contentIQ.addChannel(channelIQ);
         }
         return channelIQ;
     }
 
+    private static Integer objectToInteger(Object o)
+    {
+        Integer i;
+
+        if (o instanceof Integer)
+        {
+            i = (Integer) o;
+        }
+        else if (o instanceof Number)
+        {
+            i = ((Number) o).intValue();
+        }
+        else
+        {
+            i = Integer.valueOf(o.toString());
```

Comment:
do we want to swallow the possible exception/give a default value here? or is letting it bubble up intentional?

Reply:
i'm surprised the method didn't need a `throws ...`

Reply:
It's a RuntimeException. I don't think we want to swallow it -- if we get invalid JSON parsing should fail.